common/scripts/base.sh

```diff
@@ -89,23 +89,27 @@ function undeploy() {
   echoBold "Removing Marathon application: ${1}..."
   if ! dcos marathon app remove $1; then
     echoError "Failed to undeploy ${1}. Non-zero exit code returned from DCOS CLI"
+    return 1
   fi
   echoSuccess "Successfully undeployed ${1}"
   return 0
 }
 
-# Check whether given service port is open via marathon-lb
-# $1 - Marathon application ID
-# $2 - service port
+# Check whether given service port is open
+# $1 -Marathon application id
+# $2 -service port
 function waitUntilServiceIsActive() {
-  marathon_lb_host_ip=$(dcos marathon app show marathon-lb | $mesos_artifacts_home/common/scripts/get-marathon-lb-host.py)
-  echoBold "Waiting for ${1} to launch on ${marathon_lb_host_ip}:${2}..."
-  while ! nc -w3 $marathon_lb_host_ip $2 >/dev/null; do
-    echoBold "Waiting for ${1} to launch on ${marathon_lb_host_ip}:${2}..."
+  while ! dcos marathon app show $1 | python $mesos_artifacts_home/common/scripts/get-host-ip.py $1; do
```

Comment:
Shall we put a timeout count for all waiting points? No need to check the System time stamp. Just a simple retry count check would do. Otherwise the script will get stuck here indefinitely.